accord-core/src/main/java/accord/impl/InMemoryCommandStore.java

```diff
@@ -367,19 +369,22 @@ protected void updatedRedundantBefore(SafeCommandStore safeStore, RedundantBefor
             }
         }
         TxnId clearProgressLogBefore = unsafeGetRedundantBefore().minShardAndLocallyAppliedBefore();
-        List<TxnId> clearing = ((DefaultProgressLog) progressLog).activeBefore(clearProgressLogBefore);
-        for (TxnId txnId : clearing)
-        {
-            GlobalCommand globalCommand = commands.get(txnId);
-            Invariants.require(globalCommand != null && !globalCommand.isEmpty());
-            Command command = globalCommand.value();
-            StoreParticipants participants = command.participants().filter(LOAD, safeStore, txnId, command.executeAtIfKnown());
-            Cleanup cleanup = Cleanup.shouldCleanup(FULL, txnId, command.executeAtIfKnown(), command.saveStatus(), command.durability(), participants, unsafeGetRedundantBefore(), durableBefore());
-            Invariants.require(command.hasBeen(Applied)
-                               || cleanup.compareTo(Cleanup.TRUNCATE) >= 0
-                               || (durableBefore().min(txnId) != Universal &&
-                                      ((command.participants().stillExecutes() != null && command.participants().stillExecutes().isEmpty())
-                                      || !Route.isFullRoute(command.route()))));
+        if (progressLog instanceof DefaultProgressLog)
```

Comment:
Should we maybe at least warn otherwise? That said, probably this is fine since that's used for tests only for now.

Reply:
(but happy to if you think it is)